### ISSUE TITLE
[Feature] keep the RayCluster's prefix in the worker pods name

### ISSUE THREAD

**Issue by a-pichard** [ray-project/kuberay#2227]
### Search before asking

- [X] I had searched in the [issues](https://github.com/ray-project/kuberay/issues) and found no similar feature requirement.

### Description

Currently when worker pod name is too long the prefix is trimmed [here](https://github.com/ray-project/kuberay/blob/master/ray-operator/controllers/ray/utils/util.go#L116)

In my opinion it would be better to keep the the raycluster's prefix in the worker pod name.

For example i have
```
> kubectl get rayjobs.ray.io
NAME                       JOB STATUS   DEPLOYMENT STATUS   START TIME             END TIME   AGE
experiment-dac961a7-0611   RUNNING      Running             2024-07-09T09:51:23Z              3m41s

> kubectl get rayclusters.ray.io
NAME                                        DESIRED WORKERS   AVAILABLE WORKERS   CPUS   MEMORY   GPUS   STATUS   AGE
experiment-dac961a7-0611-raycluster-jjlc6   10                10                  42     252G     1      ready    3m36s

> kubectl get pods -l step=run
NAME                                                      READY   STATUS    RESTARTS   AGE
dac961a7-0611-raycluster-jjlc6-worker-cpu-workers-2njk5   1/1     Running   0          3m28s
dac961a7-0611-raycluster-jjlc6-worker-cpu-workers-2qkz9   1/1     Running   0          3m28s
dac961a7-0611-raycluster-jjlc6-worker-cpu-workers-4ltsn   1/1     Running   0          3m28s
dac961a7-0611-raycluster-jjlc6-worker-cpu-workers-7ghjk   1/1     Running   0          3m28s
dac961a7-0611-raycluster-jjlc6-worker-cpu-workers-llpj6   1/1     Running   0          3m28s
dac961a7-0611-raycluster-jjlc6-worker-cpu-workers-ndkhd   1/1     Running   0          3m28s
dac961a7-0611-raycluster-jjlc6-worker-cpu-workers-rp2sm   1/1     Running   0          3m28s
dac961a7-0611-raycluster-jjlc6-worker-cpu-workers-tp6hn   1/1     Running   0          3m29s
dac961a7-0611-raycluster-jjlc6-worker-cpu-workers-w2rg8   1/1     Running   0          3m28s
dac961a7-0611-raycluster-jjlc6-worker-cpu-workers-z84p2   1/1     Running   0          3m28s
experiment-dac961a7-0611-head                             1/1     Running   0          3m29s # templated on my side
experiment-dac961a7-0611-zkssv                            1/1     Running   0          18s   # submitter
```

Expected result for a worker pod would be:
```
experiment-dac961a7-0611-worker-cpu-workers-2njk5
```

I don't think that `raycluster-jjlc6` is necessary in the pod name.

the idea would be trim like this:
```
given this string
"<raycluster-name>[TRIM_HERE]" + "-" + "worker" + "-" + "<worker-group-name>[OR_TRIM_HERE]" + "-" + XXXXX
```

### Use case

Having a more robust naming strategy for the worker pods in the case of long names.
- easier to debug when you know the parent resource name
- alphabetically sorted pods should be same as alphabetically sorted rayclusters/rayjobs.

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

**Comment by a-pichard** [comment]
Also, to save some space in the pod name, do we really need the constant `worker-` ? On our side the worker group name usually contains "worker", not sure if it is generally necessary

**Comment by andrewsykim** [comment]
This seems like a reasonable ask to me. @kevin85421 can we add the v1.2 tag and prioritize this for the upcoming release?

**Comment by andrewsykim** [comment]
My biggest concern with changing naming schema is unexpected behavior when you upgrade KubeRay, we'll need to ensure we sufficiently test the upgrade paths if we change naming schemas anywhere

**Comment by andrewsykim** [comment]
I plan to work on this bug this week

**Comment by andrewsykim** [comment]
I have a draft PR for this here: https://github.com/ray-project/kuberay/pull/2267 

@a-pichard can you take a look and let me know if this would address your concerns for pod names?

**Comment by kevin85421** [comment]
Closed by #2267
